tests/unittest.py

```diff
@@ -296,7 +297,19 @@ def setUp(self) -> None:
 
         from tests.rest.client.utils import RestHelper
 
-        self.helper = RestHelper(self.hs, self.site, getattr(self, "user_id", None))
+        # HomeServer's reactor is ISynapseReactor, but for tests it should be
+        # MemoryReactorClock, which some of the internal mechanisms of tests
+        # depend on.
+        #
+        # Attempting to assert that here causes mypy to think the rest the code
+        # below the assertion to be unreachable, so just cast it. Hopefully this
+        # is true!
```

Comment:
Forgive me advertising my own hacks again, but there's `checked_cast` if you want to check this at runtime too!

Reply:
For interest, note that 3.11's stdlib offers an `assert_type`, but that's also a no-op at runtime. https://docs.python.org/3/library/typing.html?highlight=typing%20assert_type#typing.assert_type